src/Symfony/Component/Config/Util/XmlUtils.php

```diff
@@ -200,6 +200,8 @@ public static function phpize($value)
                 return false;
             case is_numeric($value):
                 return '0x' === $value[0].$value[1] ? hexdec($value) : (float) $value;
+            case preg_match('/^0x[0-9a-f]++$/i', $value):
```

Comment:
this is a possessive quantifier to forbid backtracking (because backtracking cannot allow to find another way to match for this regex anyway). this allows to match much longer strings since we won't reach the backtracking limit.
See http://php.net/manual/en/regexp.reference.repetition.php for the doc about it